src/Microsoft.ML.TimeSeries/RootCauseAnalyzer.cs

```diff
@@ -691,7 +691,7 @@ private static bool ContainsAll(Dictionary<string, Object> bigDictionary, Dictio
         {
             foreach (var item in smallDictionary)
             {
-                if (!bigDictionary.ContainsKey(item.Key) || !bigDictionary[item.Key].Equals(smallDictionary[item.Key]))
+                if (!bigDictionary.ContainsKey(item.Key) || !Object.Equals(bigDictionary[item.Key], smallDictionary[item.Key]))
```

Comment:
does it really fix the issue? `object.Equals(1, "1") is false` right?

Reply:
Of course, I have tested some special cases locally.